jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java

```diff
@@ -117,13 +141,11 @@ private static long high(String hexString) {
    */
   public static String contextAsString(JaegerSpanContext context) {
     int intFlag = context.getFlags() & 0xFF;
-    return new StringBuilder()
-        .append(context.getTraceId()).append(":")
-        .append(Utils.to16HexString(context.getSpanId())).append(":")
-        // parent=0 is special, no need to encode as full 16 characters, and more readable this way
-        .append(context.getParentId() == 0 ? "0" : Utils.to16HexString(context.getParentId())).append(":")
-        .append(Integer.toHexString(intFlag))
-        .toString();
+    return context.getTraceId() + ":"
+            + context.toSpanId() + ":"
+            // parent=0 is special, no need to encode as full 16 characters, and more readable this way
+            + (context.getParentId() == 0 ? "0" : Utils.to16HexString(context.getParentId())) + ":"
+            + Integer.toHexString(intFlag);
```

Comment:
What's the motivation for removing StringBuilder?

Reply:
Cleaner code.
I guess the reason the code is using StringBuilder is due to the assumption that when using plain concatenation the compiler / runtime will generate less optimal code. [Here's](https://stackoverflow.com/a/1532499) some relevant discussion (pertaining to Oracle JDK 6).
From Java 9 onwards the byte code generated from string concatenation is [implemented using invokedynamic](https://openjdk.java.net/jeps/280) to allow future optimisations, giving further motivation to not hardcode concatenation using StringBuilder.

Reply:
but we're not using Java 9

Reply:
I referred to the changes introduced in Java 9 just to show that plain concatenation is something that VM implementors are optimising for.
For Java 8 this code will get replaced by StringBuilder, so it's equivalent to the current implementation.